lib/syskit/data_service.rb

```diff
@@ -90,10 +90,13 @@ def find_device_attached_to(service = nil)
                     elsif driver_services.size > 1
                         raise ArgumentError, "#{self} handles more than one device, you must specify one of #{driver_services.map(&:name).sort.join(", ")} explicitely"
                     end
-                    service = driver_services.first
+                    service = driver_services.find do |driver_service|
+                        service == driver_service
+                    end
+                    
                 end
 
-                arguments["#{service.name}_dev"]
```

Comment:
Yes I made that change to not have each a single argument for each device. Where there is on argument per device type, that leads to having unassigned arguments on the task. This in turn prevents the task from being started.
This change implies that in a concrete component network each concrete TaskContext can only be driver for one particular device. I think this is a good implication and also mirrors the behavior from before this PR (where a task context could only be driver for one device type).